File: freefall-solver/src/diver_case.py
from collections import namedtuple
from dataclasses import dataclass


@dataclass
class Diver:
    weight: float
    volume_static: float
    volume_compressible: float
    drag_area: float
    drag_coefficient: float


PlotRange = namedtuple('PlotRange', 'min max num_points')


@dataclass
class PlotParameters:
    time_range: PlotRange
    depth_range: PlotRange
    velocity_range: PlotRange


@dataclass
class Scenario:
    id: int
    start_depth: float
    start_velocity: float
    extra_weight: float


@dataclass
class DiverCase:
    diver: Diver
    scenarios: list[Scenario]
    plot_parameters: PlotParameters

File: freefall-frontend/src/models/diver-case.ts
export type Diver = {
    weight: number;
    volume_static: number;
    volume_compressible: number;
    drag_area: number;
    drag_coefficient: number;
}

export type Scenario = {
    id: string;
    start_depth: number;
    start_velocity: number;
    extra_weight: number;
}

export type PlotRange = {
    min: number;
    max: number;
    num_points: number;
}

export type PlotParameters = {
    time_range: PlotRange;
    depth_range: PlotRange;
    velocity_range: PlotRange;
}

export type DiverCase = {
    diver: Diver;
    scenarios: Scenario[];
    plot_parameters: PlotParameters;
}